ACPs/77-reinventing-subnets/README.md

```diff
@@ -32,48 +32,202 @@ With no Elastic Subnets live on Mainnet, it is clear that Permissionless Subnets
 
-### Separate Subnet Validators from Primary Network Validators
+At a high-level, Subnets will now be able to manage their validator sets external to the P-Chain by setting the blockchain ID and address of their "validator manager". This validator manager is able to send Avalanche Warp Messages to the P-Chain in order to modify the validator set registered for the Subnet. In response, the P-Chain will send Avalanche Warp Message back to validator managers to notify them of validator set changes that have taken effect. Subnet Validators will no longer be required to validate the Primary Network, which removes the requirement to stake 2000 $AVAX to be a Subnet Validator. Further, Subnet Validators will only be required to sync the P-Chain (not X/C-Chain) in order to track validator set changes and support cross-Subnet communication. To maintain a Subnet Validator on the P-Chain, a continuous fee must be paid in $AVAX as specified later in this ACP.
 
-Subnet Validators will only be required to sync the P-chain (not X/C-Chain) to track any validator set changes in their Subnet and to support Cross-Subnet communication via Avalanche Warp Messaging (see "Primary Network Partial Sync" mode introduced in [Cortina 8](https://github.com/ava-labs/avalanchego/releases/tag/v1.10.8)). The lower resource requirement in this "minimal mode" will provide Subnets with greater flexibility of validation hardware requirements as operators are not required to reserve any resources for C-Chain/X-Chain operation. Since Subnet Validators are no longer required to validate or sync the Primary Network, the 2000 $AVAX requirement can be removed. To meter the number of Subnet Validators on the network, a significantly lower, continuous fee is introduced later in this ACP.
+### P-Chain Warp Message Payloads
 
-#### Bootstrapping Subnet Nodes
+To enable management of a Subnet's validator set externally to the P-Chain, Warp message verification will be added to the [`PlatformVM`](https://github.com/ava-labs/avalanchego/tree/master/vms/platformvm). For a Warp message to be considered valid by the P-Chain, at least 67% of the `sourceChainID`'s weight must have participated in the aggregate BLS signature. This is equivalent to the threshold set for the C-Chain. A future ACP may be proposed to support modification of this threshold on a per-Subnet basis.
 
-Bootstrapping a node/validator is the process of securely recreating the latest state of the blockchain locally. At the end of this process, the local state of a node/validator must be in sync with the local state of other virtuous nodes/validators. The node/validator can then verify new incoming transactions and reach consensus with other nodes/validators.
+For messages sent from the P-Chain for a given Subnet, only that Subnet's validators must be willing to provide signatures, rather than the entire Primary Network validator set. This optimization is possible because all Subnet Validators will still sync the P-Chain.
```

Comment:
```suggestion
To verify messages produced by the P-Chain, typically the entire Primary Network's validator set needs to participate in the aggregate BLS signature. However, this can cause a great deal of network bandwidth, increasing latency. An optimization can be made to only require that a quorum of the _Subnet's_ validator set needs to participate in the aggregate BLS signature for messages that are produced by the P-Chain and consumed by the Subnet.
```